python/ray/serve/_private/http_state.py

```diff
@@ -182,6 +182,13 @@ def _get_target_nodes(self) -> List[Tuple[str, str]]:
         location = self._config.location
         target_nodes = get_all_node_ids(self._gcs_client)
 
+        # Make ip address as string if it is bytes.
+        for index in range(len(target_nodes)):
+            node_id, ip_address = target_nodes[index]
+            if type(ip_address) == bytes:
+                ip_address = ip_address.decode("utf-8")
+                target_nodes[index] = (node_id, ip_address)
```

Comment:
huh, is this change required for this PR? is it because we want to write the node_ip in string to the logs now?

Reply:
yes, otherwise json dumps will raise error.
Our function/method annotation says it is `str`, yet ip address is `bytes`.

Reply:
Ah I ran into the same issue, I added the bytes decoding [here](https://github.com/ray-project/ray/blob/master/python/ray/serve/_private/utils.py#L177). You should probably keep either my change or your change here.

Reply:
nice!

Reply:
prefer where cindy has it so it's fixed everywhere we use that util

Reply:
rebased